tests/connections.py

```diff
@@ -264,6 +265,73 @@ def create_connections_until_exhaustion(
             LOG.warning("Expected a fatal crash and saw none!")
 
 
+def run_idle_timeout_tests(args):
+    test_cases = [
+        {"timeout": 5, "safe_sleeps": [2, 3], "killed_sleeps": [10]},
+        {
+            "timeout": None,
+            "safe_sleeps": [10],
+        },  # With no timeout, idle sessions are never killed
+    ]
+
+    def verbose_sleep(sleep_time):
+        slept = 0
+        step_size = 1
+        LOG.info(f"Sleeping {sleep_time}s")
+        while slept < sleep_time:
+            next_sleep = min(sleep_time - slept, step_size)
+            time.sleep(next_sleep)
```

Comment:
At the second granularity this is probably not going to matter, but sleep is _at least_, not exactly, so something that keeps track of `slept` using `time.time()` will be somewhat more accurate.